### PR TITLE
style(*): remove unused variables

### PR DESCRIPTION
**Pull request by jbedard**
... and in some cases make use of unused vars, and rename a duplicate var.

It would be nice of some of these could be caught by the build...

### CODE REVIEW

File: src/ng/templateRequest.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 'use strict';
 
-var $compileMinErr = minErr('$compile');
+var $templateRequestMinErr = minErr('$compile');
```
Comment (gkalpak) [inline src/ng/templateRequest.js]: You could also just reuse the already created `$compileMinErr`.

**Comment by gkalpak** [comment]
LGTM

File: src/ng/templateRequest.js (hunk 1 above)
Comment (jbedard) [inline src/ng/templateRequest.js]: I figured this file shouldn't depend on compile.js like that. Might want to also remove the `'$compile'` string but I figured that would be another change...

File: src/ng/directive/ngModel.js [hunk 2]
```diff
@@ -265,9 +265,9 @@ var NgModelController = ['$scope', '$exceptionHandler', '$attrs', '$element', '$
       };
       ngModelSet = function($scope, newValue) {
         if (isFunction(parsedNgModel($scope))) {
-          invokeModelSetter($scope, {$$$p: ctrl.$modelValue});
+          invokeModelSetter($scope, {$$$p: newValue});
         } else {
-          parsedNgModelAssign($scope, ctrl.$modelValue);
+          parsedNgModelAssign($scope, newValue);
```
Comment (petebacondarwin) [inline src/ng/directive/ngModel.js]: @lgalfaso are these two changes correct?

Comment (lgalfaso) [inline src/ng/directive/ngModel.js]: this is fine as `ngModelSet` is only called from
https://github.com/jbedard/angular.js/blob/unused/src/ng/directive/ngModel.js#L755

**Comment by petebacondarwin** [comment]
In that case LGTM

**Comment by Narretz** [comment]
jshint has an unused rule. Let's see what happens when we use that.

**Comment by jbedard** [comment]
Currently that doesn't work very well because of how variables are shared across files and jshint is run on the src files. If jshint was instead run on the resulting angular.js it would work, or maybe if shared variables are marked as globals or the rule is disabled for those vars?

**Comment by Narretz** [comment]
Many files use jshint global already. Some variables actually show up as undefined if they are not marked global. But maybe that's just my personal setup.

**Comment by Narretz** [comment]
Ah, I confused unused with undef ... So yes, directives, providers, services all trigger unused because they are only used in AngularPublic.js. This file already has a global definition, it's just that a lot of functions and variables are missing.
And for some reason, the angular api functions are also marked as unused, even though they are marked global in the src\jshint file

**Pull request closed without merging**